article-analysis-project/article-processing.py

```diff
@@ -0,0 +1,207 @@
+            api_key='66982c609ec00aba67eee6fb146f6210',
+            timeout=30.0)
+
+        #HTML Processing
+        fp = urllib.request.urlopen(URL)
+        mybytes = fp.read()
+
+        mystr = mybytes.decode("utf8")
+        fp.close()
+        soup = BeautifulSoup(mystr, 'lxml')
+
+        try:       
+            #byline = soup.find("div", class_="Byline__Author").text  #Byline for ABC
+            byline = soup.find("div", class_="founders-cond f6 lh-none").text #Byline for NBC
+        except:
+            byline = ''
+
+        authors = author_from_byline(byline)
+        author_gender = gender_of_author(authors)
+        sources= extract_sources_displacy(sentences)
+        source_genders = source_gender(sources, sentences)
+
+        count_he = source_genders.count('male')
+        count_she = source_genders.count('female')
+
+        complete_row = [article.title, authors, author_gender, sources, count_he, count_she, URL] #Create row for article
+        complete.append(complete_row)
+        #write_to_json(article.title, author, gender_of_author, sources, count_he, count_she)
+
+    complete_np = np.array(complete)
+    df = pd.DataFrame(complete_np, columns = ['Title', 'Author', 'Gender of Author', 'Sources', 'Male Sources', 'Female Sources', 'url'])
+    df.to_csv('complete_article_data_nbc.csv')
+
+def gender_of_author(authors):
+    genderize = Genderize(
+        user_agent='Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36',
+        api_key='66982c609ec00aba67eee6fb146f6210',
+        timeout=30.0)
+    #Return first name
+    first_names = []
+    for author in authors:
+        first_names.append(author.split()[0])
+    
+    genders_full = genderize.get(first_names)
+    genders =[]
+    for d in genders_full:
+        genders.append(d['gender'])
+    return genders
+
+def author_from_byline(byline):
+    nlp = spacy.load('en_core_web_sm') #Load english
+    docx = nlp(byline.lower())
+    authors = []
+    for ent in docx.ents:
+        if ent.label_ == 'PERSON':
+            authors.append(ent.text)
+    return authors
+    
+def write_to_json(title, authors, gender_of_author, sources, count_male, count_female):
+     #Write to JSON
+        # Data to be written
+        dictionary ={ 
+            "title" : article.title, 
+            "Author" : authors, 
+            "Gender of Author": gender_of_author,
+            "Sources": np.array_str(sources),
+            "# of male sources" : count_male, 
+            "# of female sources" : count_female
+        } 
+    
+        # Serializing json  
+        json_object = json.dumps(dictionary, indent = 4) 
+    
+        # Writing to sample.json 
+        with open("example.json", "w") as outfile: 
+            outfile.write(json_object) 
+
+def extract_sources_displacy(sentences):
+    nlp = spacy.load('en_core_web_sm') #Load english
```

Comment:
I'd throw this in a well named constant (which also makes the comment unnecessary)